Tetrahedral_remeshing/examples/Tetrahedral_remeshing/mesh_and_remesh_polyhedral_domain_with_features.cpp

```diff
@@ -0,0 +1,70 @@
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+
+#include <CGAL/Mesh_triangulation_3.h>
+#include <CGAL/Mesh_complex_3_in_triangulation_3.h>
+#include <CGAL/Mesh_criteria_3.h>
+
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/Polyhedral_mesh_domain_with_features_3.h>
+#include <CGAL/make_mesh_3.h>
+
+#include <CGAL/tetrahedral_remeshing.h>
+
+// Domain
+typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
+typedef CGAL::Surface_mesh<K::Point_3> Polyhedron;
+typedef CGAL::Polyhedral_mesh_domain_with_features_3<K, Polyhedron> Mesh_domain;
+
+// Triangulation for Meshing
+typedef CGAL::Mesh_triangulation_3<Mesh_domain, CGAL::Default, CGAL::Default>::type Tr;
+typedef CGAL::Mesh_complex_3_in_triangulation_3<
+  Tr, Mesh_domain::Corner_index, Mesh_domain::Curve_index> C3t3;
+
+// Criteria
+typedef CGAL::Mesh_criteria_3<Tr> Mesh_criteria;
+
+// Triangulation for Remeshing
+typedef CGAL::Triangulation_3<typename Tr::Geom_traits,
+  typename Tr::Triangulation_data_structure> Triangulation_3;
+
+// To avoid verbose function and named parameters call
+using namespace CGAL::parameters;
+
+int main(int argc, char* argv[])
+{
+  const char* fname = (argc > 1) ? argv[1] : "data/fandisk.off";
+  std::ifstream input(fname);
+  Polyhedron polyhedron;
+  input >> polyhedron;
+  if (input.fail()) {
+    std::cerr << "Error: Cannot read file " << fname << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  if (!CGAL::is_triangle_mesh(polyhedron)) {
+    std::cerr << "Input geometry is not triangulated." << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // Create domain
+  Mesh_domain domain(polyhedron);
+
+  // Get sharp features
+  domain.detect_features();
+
+  // Mesh criteria
+  Mesh_criteria criteria(edge_size = 0.025,
+    facet_angle = 25, facet_size = 0.05, facet_distance = 0.005,
+    cell_radius_edge_ratio = 3, cell_size = 0.05);
+
+  // Mesh generation
+  C3t3 c3t3 = CGAL::make_mesh_3<C3t3>(domain, criteria);
+
+  Triangulation_3 tr = CGAL::convert_to_triangulation_3(c3t3);
+
```

Comment:
```suggestion
  Triangulation_3 tr = CGAL::convert_to_triangulation_3(std::move(c3t3));

```

in my opinion, that is not the job of the CGAL reference manual to teach users about C++. But we can encourage good practice in our examples.

Reply:
both options can be present (one being commented) and some comment should be added to make it clear what both options imply.